### PR TITLE
[mold] Add latest version 1.4.1

### PR DESCRIPTION
**Pull request by ericriff**
Mold has moved to CMake on its latest release, so this PR makes the recipe a bit convoluted. 
Should we move the old recipe to a `1.3` folder and keep it as a different recipe? Deprecate 1.3 and keep only 1.4+? Keep both build systems on the same recipe as this PR proposes?

### CODE REVIEW

**Comment by ericriff** [comment]
Well, I'll decline this. The recipe is not compatible with the newer version and I can't make it build locally, so working on this will be tedious for me. 
The latest version now supports CMake, we should use it instead of plain makefiles.
I'll reopen this if I get the time to play around with a docker container with newer toolchains.

**Comment by AndreyMlashkin** [comment]
@ericriff, I made it work with gcc10:
https://github.com/conan-io/conan-center-index/pull/12782

**Comment by AndreyMlashkin** [comment]
I would just create a new recipe parallel to the exisiting one and use cmake there

**Comment by ericriff** [comment]
@AndreyMlashkin cool. This PR is just an experiment and needs a lot of cleanup. I'm not sure how to handle the change in build systems. Maybe we can deprecate 1.3.x? Or keep using makefiles on 1.4?

File: recipes/mold/all/conanfile.py [hunk 1]
```diff
@@ -64,26 +70,57 @@ def _patch_sources(self):
...
-            autotools = AutoToolsBuildEnvironment(self)
-            autotools.make(target="mold", args=['SYSTEM_TBB=1', 'SYSTEM_MIMALLOC=1'])
+        if Version(self.version) < "1.4":
+            self._patch_sources()
+            with files.chdir(self, self._source_subfolder):
+                autotools = AutoToolsBuildEnvironment(self)
+                autotools.make(target="mold", args=['SYSTEM_TBB=1', 'SYSTEM_MIMALLOC=1'])
+        else:
+            # Error out if ZLIB is not found, we want to be predictable
+            files.replace_in_file(self, "source_subfolder/CMakeLists.txt", "find_package(ZLIB QUIET)", "find_package(ZLIB REQUIRED)")
```
Comment (ericriff) [inline recipes/mold/all/conanfile.py:104]: I'll move this to a proper helper function once we agree on the topics mentioned on the PR description

**Comment by AndreyMlashkin** [comment]
I am using a cmake chain here:
https://github.com/conan-io/conan-center-index/pull/12881

I think, you can close this PR

**Comment by ericriff** [comment]
Closed in favor of another PR.

**Pull request closed without merging**